src/App.tsx

```diff
@@ -1,26 +1,163 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './commponents/Header';
+import { Footer } from './commponents/Footer';
+import { Errors } from './commponents/Errors';
+import { TodoCard } from './commponents/TodoCard';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorType } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<Status>(Status.All);
+  const [errorType, setErrorType] = useState<string>('');
+  const [newTodo, setNewTodo] = useState<Todo | null>(null);
+  const [deleteItemTodo, setDeleteItemTodo] = useState(NaN);
+  const [loadedDelete, setLoadedDelete] = useState(false);
+  const [todoTask, setTodoTask] = useState('');
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const timeoutId = setTimeout(() => setErrorType(''), 3000);
+
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+      } catch {
+        setErrorType(ErrorType.UnableToLoad);
+        clearTimeout(timeoutId);
+      }
+
+      return () => clearTimeout(timeoutId);
```

Comment:
You should do this `return` from `useEffect`, not from `function`
It should work like `ComponentWillUnmount`